src/Makefile.bench.include

```diff
@@ -52,13 +52,13 @@ nodist_bench_bench_bitcoin_SOURCES = $(GENERATED_BENCH_FILES)
 bench_bench_bitcoin_CPPFLAGS = $(AM_CPPFLAGS) $(BITCOIN_INCLUDES) $(EVENT_CFLAGS) $(EVENT_PTHREADS_CFLAGS) -I$(builddir)/bench/
 bench_bench_bitcoin_CXXFLAGS = $(AM_CXXFLAGS) $(PIE_FLAGS)
 bench_bench_bitcoin_LDADD = \
+  $(LIBTEST_UTIL) \
```

Comment:
Perhaps? Although I think GNU bfd ld still has requirements on library ordering so we should probably keep it...

Reply:
Yes, but only to default-enforce the (very sane imo) rule of no circular dependencies.

IIRC `preserve-dup-deps` was added at one point because we had a real circular dependency. If either of:
- There was never actually a circular dependency and the flag was added as an overkill way to solve the ordering problem
- There was but no longer is a circular dependency

then imo the flag should be removed to put us back in the "strict ordering" mode, which would be helpful for pointing out accidental _new_ circular dependencies.

But if I'm wrong about those two conditions, just ignore me :)

Reply:
Okay I see! Let's do this as a followup.